src/components/ChatEntry.js

```diff
@@ -1,22 +1,40 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp';
+
+const ChatEntry = ({ id, sender, body, timeStamp, onLikeChange }) => {
+  const [liked, setLiked] = useState(false);
+
+  const handleLike = () => {
+    const newLiked = !liked;
+    setLiked(newLiked);
+    onLikeChange(id, newLiked);
```

Comment:
👀 Our local component here shouldn't need to know how to toggle whether this message is liked. Rather, this component should only pass the id of the message whose like button was clicked to the callback, and _that_ logic should be responsible for determining what the new liked status for the message should be (the liked count could be calculated on the next render). This would help ensure we have a single source of truth (the list of messages), from which we can draw the entire UI.
```js
  const handleLikeToggle = () => {
    // assume the callback function is passed in as onLikeClicked (reporting the click, not a change)
    props.onLikeClicked(id);
  }

  // in App.js
  const handleLikeClicked = (id) => {
    // copy the list of message stored in the App state
    // copy the message indicated by the id
    // flip the liked status for the message
    // update the App state tracking the messages
  };

  // later in App.js
  <ChatLog entries={messages} onLikeClicked={handleLikeClicked} />
```